client/src/components/VtkViewer.vue

```diff
@@ -195,54 +195,19 @@ export default {
       }
       return currClosest;
     },
-    findMaxLocation(matrix) {
-      matrix = matrix.map((value) => Math.abs(value));
-      let maximum = -1;
-      let location = [-1, -1];
-      for (let r = 0; r < 3; r += 1) {
-        for (let c = 0; c < 3; c += 1) {
-          const index = r * 3 + c;
-          if (matrix[index] > maximum) {
-            maximum = matrix[index];
-            location = [r, c];
-          }
-        }
-      }
-      return location;
-    },
-    zeroRowColumn(matrix, r, c) {
-      for (let i = 0; i < 3; i += 1) {
-        const index = i * 3 + c;
-        matrix[index] = 0;
-      }
-      for (let i = 0; i < 3; i += 1) {
-        const index = r * 3 + i;
-        matrix[index] = 0;
-      }
-      return matrix;
-    },
-    normalizedAxis(oldAxis) {
-      let orientation = this.representation.getInputDataSet().getDirection().slice();
-      const newMatrix = [0, 0, 0, 0, 0, 0, 0, 0, 0];
-      for (let i = 0; i < 3; i += 1) {
-        const currentMaxLocation = this.findMaxLocation(orientation);
-        const index = currentMaxLocation[0] * 3 + currentMaxLocation[1];
-        newMatrix[index] = orientation[index];
-        orientation = this.zeroRowColumn(orientation, currentMaxLocation[0], currentMaxLocation[1]);
-      }
-      const axisMapping = {
-        x: newMatrix.slice(0, 3),
-        y: newMatrix.slice(3, 6),
-        z: newMatrix.slice(6, 9),
-      };
-      const absoluteTargetColumn = axisMapping[oldAxis].map((value) => Math.abs(value));
-      const maxLocation = absoluteTargetColumn.indexOf(Math.max(...absoluteTargetColumn));
-      const flip = Math.sign(axisMapping[oldAxis][maxLocation]);
-      const axes = ['x', 'y', 'z'];
-      return {
-        axis: axes[maxLocation],
-        flip: flip < 0,
-      };
+    trueAxis(axisName) {
+      const orientation = this.representation.getInputDataSet().getDirection();
+      const axisNumber = VIEW_ORIENTATIONS[axisName].axis;
+      const axisOrientation = orientation.slice(
+        axisNumber * 3, axisNumber * 3 + 3,
+      ).map(
+        (val) => Math.abs(val),
+      );
+      const axisOrdering = ['x', 'y', 'z'];
+      const trueAxis = axisOrdering[
+        axisOrientation.indexOf(Math.max(...axisOrientation))
```

Comment:
Wouldn't this select some axis twice in certain oblique cases? What does `oblique2` test case look like now? Will this interact with the expected fix of https://github.com/Kitware/vtk-js/issues/2191?

Reply:
Good point. It may not be a problem once that bug is fixed, but it wouldn't hurt to come up with a more foolproof way of selecting the true axis. I'm not sure the best way. What do you recommend?

I recorded the current behavior of CTChestOblique2rot in the following video:

https://user-images.githubusercontent.com/44912689/147777237-94a4e920-325e-481e-9c63-a1948099e8f1.mov

Reply:
I guess it was like this before this simplification? As long as we have the more complicated code in history or in a branch, we should move forward with this. When vtk.js bug gets fixed, we should revisit this. Then either revive or totally discard the more complicated code.